### PR TITLE
Remove forced override of package contents

### LINKED ISSUE #1155: how to use `System.Collections.Immutable` v`8.0.0` with `Microsoft.CodeAnalysis.Testing.ReferenceAssemblies.Net.Net60`
our code-analyzer-tests use `Microsoft.CodeAnalysis.CSharp.CodeFix.Testing.MSTest`. We now had to upgrade `System.Collections.Immutable`  to `8.0.0` for all Assemblies. Whenever we now want to use these assemblies to run within tests (because we have tests specifically for Classes inside our assemblies) we get an Exception:
```txt
// error CS1705: Assembly 'Komsa' with identity 'Komsa, Version=2.23.1.0, Culture=neutral, PublicKeyToken=null' uses 'System.Collections.Immutable, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' which has a higher version than referenced assembly 'System.Collections.Immutable' with identity 'System.Collections.Immutable, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
DiagnosticResult.CompilerError("CS1705").WithArguments("Komsa", "Komsa, Version=2.23.1.0, Culture=neutral, PublicKeyToken=null", "System.Collections.Immutable, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a", "System.Collections.Immutable", "System.Collections.Immutable, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"),
```

adding
```cs
new Verifier.Test()
{
  TestState =
  {
    AdditionalReferences = { typeof(ImmutableArray).Assembly  },
  }
}
```
results in the following:
```txt
Actual diagnostic:
    // /0/Test0.cs(13,5): error CS0433: The type 'ImmutableArray<T>' exists in both 'System.Collections.Immutable, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' and 'System.Collections.Immutable, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
DiagnosticResult.CompilerError("CS0433").WithSpan(13, 5, 13, 27).WithArguments("System.Collections.Immutable, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a", "System.Collections.Immutable.ImmutableArray<T>", "System.Collections.Immutable, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"),
```

adding the NugetPackage like this does not change anything (in either above case):
```cs
new Verifier.Test()
{
  ReferenceAssemblies = ReferenceAssemblies.Net.Net60 // used always
    .AddPackages([new PackageIdentity("System.Collections.Immutable", "8.0.0")]) // new
}
```

Our verifiers are essentially a "thin wrapper" around `CodeFixVerifier<TAnalyzer, TCodeFix, TTest, TVerifier>` and Verifier.Test around `CSharpCodeFixTest<TAnalyzer, TCodeFix, TVerifier>`, to allow for more testing and better migrating from our old tests, and as a workaround for other bugs in the past (ie tests not being able to handle 2-space indents correctly). We do not interfere with Roslyn in this subclass, we "only extend/append" (editorconfig-spacing) on what the base implementation provides. Thus I don't see us being the reason for "something not working"

### PR DESCRIPTION
**Pull request by sharwell**
Fixes #1155

### CODE REVIEW

File: src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/ReferenceAssemblies.cs [hunk 1]
```diff
@@ -460,14 +460,51 @@ private async Task<ImmutableArray<MetadataReference>> ResolveCoreAsync(string la
...
+                            //
+                            // Microsoft.NETCore.App.Ref 6.0.0 provides System.Collections.Immutable in the net6.0 folder
+                            // System.Collections.Immutable 8.0.0 provides System.Collections.Immutable in the net6.0 folder
+                            //
+                            // In this example, the Microsoft.NETCore.App.Ref package is resolved first, so by taking
+                            // the last net6.0 assembly, we ensure the assembly from System.Collections.Immutable 8.0.0
+                            // is resolved.
+                            if (Equals(GetFrameworkNameFromPath(assembliesByPrecedence[0]), GetFrameworkNameFromPath(assembliesByPrecedence[i])))
+                            {
+                                assembliesToRemove.Add(assembliesByPrecedence[i - 1]);
```
Comment (ToddGrun) [inline src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/ReferenceAssemblies.cs:487]: >assembliesToRemove [](http://example.com/codeflow?start=32&length=18)

Why not just directly remove from resolvedAssemblies?

Comment (ToddGrun) [reply src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/ReferenceAssemblies.cs:487]: (I assume it's the GroupBy usage, but if that were the case, I would have thought just realizing that result would have been easier)

File: src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/ReferenceAssemblies.cs [hunk 2]
```diff
@@ -460,14 +460,51 @@ private async Task<ImmutableArray<MetadataReference>> ResolveCoreAsync(string la
...
+                            // be a case where the reference assembly package provides the assembly and a newer version
+                            // is provided explicitly. For example:
+                            //
+                            // Microsoft.NETCore.App.Ref 6.0.0 provides System.Collections.Immutable in the net6.0 folder
+                            // System.Collections.Immutable 8.0.0 provides System.Collections.Immutable in the net6.0 folder
+                            //
+                            // In this example, the Microsoft.NETCore.App.Ref package is resolved first, so by taking
+                            // the last net6.0 assembly, we ensure the assembly from System.Collections.Immutable 8.0.0
+                            // is resolved.
+                            if (Equals(GetFrameworkNameFromPath(assembliesByPrecedence[0]), GetFrameworkNameFromPath(assembliesByPrecedence[i])))
```
Comment (ToddGrun) [inline src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/ReferenceAssemblies.cs:485]: >GetFrameworkNameFromPath(assembliesByPrecedence[0]) [](http://example.com/codeflow?start=39&length=51)

as this is invariant, consider moving it outside the loop

File: src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/ReferenceAssemblies.cs (hunk 1 above)
Comment (sharwell) [reply src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/ReferenceAssemblies.cs:487]: ➡️ I'm going to keep the current form for ease of debugging. Will add a comment in the code explaining that.

File: src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/ReferenceAssemblies.cs (hunk 2 above)
Comment (sharwell) [reply src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/ReferenceAssemblies.cs]: ➡️ Moved to single evaluation of `GetFrameworkNameFromPath`

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Pull request merged**